react/features/filmstrip/reducer.js

```diff
@@ -116,6 +117,14 @@ ReducerRegistry.register(
                 ...state,
                 horizontalViewDimensions: action.dimensions
             };
+        case SET_REMOTE_PARTICIPANTS: {
+            const { visibleParticipantsStartIndex: startIndex, visibleParticipantsEndIndex: endIndex } = state;
+
+            state.remoteParticipants = action.participants;
+            state.visibleParticipants = state.remoteParticipants.slice(startIndex, endIndex + 1);
+
+            return state;
```

Comment:
I believe you may need to return `{ ...state }`. Redux will check the returned state by reference and trigger the subscribers only if the reference is changed. 

In `PARTICIPANT_JOINED` I didn't do that by mistake and it was luckily working because from the participant's reducer we were returning new reference for the state.